RELEASE_PROCESS_OPENSEARCH.md

```diff
@@ -0,0 +1,313 @@
+
+### Version Increment
+
+Versions are incremented as soon as development starts on a given version to avoid confusion. Following is the example that depicts the version increment.
+
+* OpenSearch: `main` = 2.0.0, `1.x` = 1.1.0, and `1.0` = 1.0.0
+* common-utils: `main` = 1.1.0.0, `1.0` = 1.0.0.0
+
+#### Version Increment Core
+
+To ensure the version incrementation process is handled correctly, it is important to follow up with the core team. Currently, the version incrementation is being done manually by an individual from the core repositories. Sample OpenSearch Version Increment [PR](https://github.com/opensearch-project/OpenSearch/pull/7864/).
+
+#### Release Branch
+
+Not applicable for patch release. This step requires that every team participating in a release has their release branch created for the corresponding release by the date listed on this step. The distribution build job will also start using the release branch to create release candidate instead of `.x` branches.
+
+#### Plugins Version Increment
+
+The next step is to identify the pending plugin version increment pull requests. The creation of plugin version increment pull requests is automated with workflows running in the build repo [OpenSearch Version Increment Workflow](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/os-increment-plugin-versions.yml),
+[OpenSearch Dashboards Version Increment Workflow](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/osd-increment-plugin-versions.yml). Sample plugin version increment [PR](https://github.com/opensearch-project/job-scheduler/pull/398).
+
+The goal is to get these pull requests merged so that all the plugins are in sync with the core version. It's not always just raising a pull request and getting it merged. There are CI checks for every repo that get triggered when a pull request is raised. These CI checks test against the dependencies. Therefore, before the version increment pull requests are merged, the release manager needs to ensure that the version increment is done for dependencies, built, and pushed to Maven and S3. This is a key step for the release manager to coordinate with the plugin teams, address the dependencies, and get the version increment pull requests merged.
+
+### Distribution Build
+
+Refer the [Build Workflows](#build-workflows) section to get the details about the distribution build jobs. This section covers how to sequentially
+handle the distribution build job during the release.
+
+Following is the order of execution for the distribution build:
+
+##### OpenSearch
+```
+OpenSearch 
+OpenSearch + common-utils + job-scheduler 
+OpenSearch + common-utils + job-scheduler + performance-analyzer 
+OpenSearch + common-utils + job-scheduler + performance-analyzer + security 
+OpenSearch + common-utils + job-scheduler + ml-commons + performance-analyzer + security 
+All components (which are ready after completion of version increment)
```

Comment:
Outdated as the maven publishing is already on individual repo github actions.

Reply:
Yes but is it good to pick them from local maven repo instead of the published snapshot repo? that way the distribution is build and assembled from the artifacts build at point in time and fetched from the workspace.

Reply:
Not really, the build is never the problem if plugins are merged with their PRs.
Previously the issue is the jenkins run is required to publish maven jars, so plugin repos PR can pass the checks to merge.

Reply:
Thanks added a NOTE peter.